controller/controller_server/csi_controller_server.py

```diff
@@ -136,37 +123,10 @@ def CreateVolume(self, request, context):  # pylint: disable=too-many-branches
                 res = utils.generate_csi_create_volume_response(volume, source_type)
                 logger.info("finished create volume")
                 return res
-
-        except (controller_errors.IllegalObjectName, controller_errors.SpaceEfficiencyNotSupported,
-                controller_errors.PoolDoesNotExist, controller_errors.PoolDoesNotMatchCapabilities,
-                controller_errors.PoolParameterIsMissing, controller_errors.ExpectedSnapshotButFoundVolumeError) as ex:
-            logger.exception(ex)
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.ObjectNotFoundError as ex:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            context.set_details(ex.message)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.PermissionDeniedError as ex:
-            context.set_code(grpc.StatusCode.PERMISSION_DENIED)
-            context.set_details(ex.message)
-            return csi_pb2.CreateVolumeResponse()
+        except controller_errors.InvalidArgumentError as ex:
+            handle_exception(ex, context, grpc.StatusCode.INVALID_ARGUMENT, csi_pb2.CreateVolumeResponse)
         except controller_errors.VolumeAlreadyExists as ex:
-            context.set_details(ex.message)
-            context.set_code(grpc.StatusCode.ALREADY_EXISTS)
-            return csi_pb2.CreateVolumeResponse()
-        except controller_errors.NotEnoughSpaceInPool as ex:
```

Comment:
I think this `except` block [was added](https://github.com/IBM/ibm-block-csi-driver/pull/251/files#diff-1d72325bc9ed2e0861808fe985ff7e74edfbba4533a5a71f12b8fd498bb2647c) here [by mistake](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#createvolume-errors) - so it shouldn't be in the common exceptions.
the status code for this exception is only mandatory in `CreateSnapshot`, optional in `ControllerExpandVolume`, and is a bug in `CreateVolume`.
it can be fixed as part of https://github.com/IBM/ibm-block-csi-driver/pull/349

Reply:
[Done](https://github.com/IBM/ibm-block-csi-driver/pull/349/commits/648349dc35b82ec9070fd807ef16b06628df1d40)